src/Compilers/Core/Portable/Operations/BasicBlock.cs

```diff
@@ -84,28 +84,22 @@ internal void RemovePredecessor(BasicBlock block)
             _predecessors.Remove(block);
         }
 
-        [Flags]
-        public enum BranchFlags
+        public enum BranchKind
         {
+            // Keep 0 value unused to detect an uninitialized state
             Regular = 0x01,
-
-            /// <summary>
-            /// This is the only flag that can be combined with other values.
-            /// </summary>
-            Error = 0x02,
-
-            Return = 0x04,
-            StructuredExceptionHandling = 0x08,
-            ProgramTermination = 0x10,
-            Throw = 0x20,
-            ReThrow = 0x40,
+            Return,
```

Comment:
Since these are part of public API, perhaps it's better to keep the values explicit.

Reply:
> Since these are part of public API, perhaps it's better to keep the values explicit.

I don't think this is necessary, even when enum is public. We know how enums "work" an will make sure to not change the underlying values. The public API checker also flags changes to the values, whether implicit or explicit